src/battle_util.c

```diff
@@ -7307,6 +7307,19 @@ static u32 CalcAttackStat(u16 move, u8 battlerAtk, u8 battlerDef, u8 moveType, b
             atkStage = gBattleMons[battlerDef].statStages[STAT_SPATK];
         }
     }
+    if (gBattleMoves[move].effect == EFFECT_BODY_PRESS)
+    {
+        if (IS_MOVE_PHYSICAL(move))
+        {
+            atkStat = gBattleMons[battlerAtk].defense;
+            atkStage = gBattleMons[battlerAtk].statStages[STAT_DEF];
+        }
+        else
+        {
+            atkStat = gBattleMons[battlerAtk].spDefense;
+            atkStage = gBattleMons[battlerAtk].statStages[STAT_SPDEF];
+        }
```

Comment:
It doesn't look like Body Press uses the Sp. Def instead of the Sp. Atk in the official games, so that should go away imo, as we strive to remain as faithful as we can whenever it's possible.

Reply:
Sure, makes sense! I only included that as Foul Play has a similar effect (making it special makes it use the target's special attack) added in the battle engine and it seemed like it would be a nice convenience thing.